File: twidere/src/main/java/org/mariotaku/twidere/model/ParameterizedExpression.java
package org.mariotaku.twidere.model;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import org.mariotaku.sqliteqb.library.Expression;

/**
 * Created by mariotaku on 16/6/22.
 */
public class ParameterizedExpression {
    Expression expression;
    String[] parameters;

    public ParameterizedExpression(@NonNull Expression expression, @Nullable String[] parameters) {
        this.expression = expression;
        this.parameters = parameters;
    }

    public String[] getParameters() {
        return parameters;
    }

    public String getSQL() {
        return expression.getSQL();
    }
}

File: twidere/src/main/kotlin/org/mariotaku/twidere/model/ParameterizedExpression.kt
/*
 *             Twidere - Twitter client for Android
 *
 *  Copyright (C) 2012-2017 Mariotaku Lee <mariotaku.lee@gmail.com>
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.mariotaku.twidere.model

import org.mariotaku.sqliteqb.library.Expression

/**
 * Created by mariotaku on 16/6/22.
 */
data class ParameterizedExpression(var expression: Expression, val parameters: Array<String>) {
    val sql: String
        get() = expression.sql

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ParameterizedExpression

        if (expression != other.expression) return false
        if (!parameters.contentEquals(other.parameters)) return false

        return true
    }

    override fun hashCode(): Int {
        var result = expression.hashCode()
        result = 31 * result + parameters.contentHashCode()
        return result
    }
}
